src/launchpad/insights/insight.py

```diff
@@ -1,20 +1,19 @@
 from abc import abstractmethod
 from dataclasses import dataclass
-from typing import Protocol, TypeVar
+from typing import Protocol, Sequence, TypeVar
 
-from ..models.apple import AppleAppInfo, MachOBinaryAnalysis
-from ..models.common import FileAnalysis
-from ..models.treemap import TreemapResults
+from launchpad.models.common import BaseAppInfo, BaseBinaryAnalysis, FileAnalysis
+from launchpad.models.treemap import TreemapResults
 
 T_co = TypeVar("T_co", covariant=True)
 
 
 @dataclass
 class InsightsInput:
-    app_info: AppleAppInfo
+    app_info: BaseAppInfo
     file_analysis: FileAnalysis
     treemap: TreemapResults | None
-    binary_analysis: list[MachOBinaryAnalysis]
+    binary_analysis: Sequence[BaseBinaryAnalysis] | None = None
```

Comment:
Can this get rid of the `None`? We can use an empty list in that case. Also can this change back to `List`

Reply:
i got rid of the None.
Regarding the `List`, I’m not python expert but this was the error I got when I tried to use `List`. https://github.com/getsentry/launchpad/pull/51#discussion_r2158834575